README.md
# Mod-FictionalPotato
A tuber that which does not exist in reality


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.14.22
Date: 03. 02. 2017
  Bugfixes:
    - Limit maximum texture size the game will try to use to 16384x16384 pixels.
    - Fixed desync related to building of rail signals. (https://forums.factorio.com/37853)
    - Fixed multiplayer map download getting stuck at 100% when using some broken routers.(https://forums.factorio.com/32646)
---------------------------------------------------------------------------------------------------
Version: 0.14.21
Date: 24. 11. 2016
  Bugfixes:
    - Fixed that the game could crash when a game disappeared from the matching server before its details were requested. (https://forums.factorio.com/35893)
    - Fixed that numpad numbers didn't work for any game controls. (https://forums.factorio.com/37005)
    - Fixed crash when merging forces while a player in one the force to be removed was crafting something. (https://forums.factorio.com/37009)
    - Fixed game would be stuck in main menu if Join Game on Steam failed for some reason.
    - Fixed possible save corruption when roboport was destroyed while robot was repairing it. (https://forums.factorio.com/37647)
---------------------------------------------------------------------------------------------------
Version: 0.14.20
Date: 09. 11. 2016
  Bugfixes:
    - Fixed LuaSurface::can_place_entity didn't work for tile ghosts. (https://forums.factorio.com/35651)
    - Fixed curved rails woudln't render correctly as enemy forces.
    - Fixed searching in the technology GUI wouldn't work in some cases. (https://forums.factorio.com/35825)
    - Fixed that the config option other.use_version_filter_in_browse_games_gui defaulted to false on linux. (https://forums.factorio.com/35451)
    - Fixed "E" (close GUI) couldn't be used when numeric-input fields were focused. (https://forums.factorio.com/35807)
    - Fixed that disabling recipe groups in the settings disabled subgroups as well.
    - Fixed that the changelog GUI allowed editing the read-only text. (https://forums.factorio.com/35788)
    - Fixed (again) that the natural signal direction wasn't picked primarily when 2 way signal and normal signal can be built on the same spot.
    - Fixed that deactivated belts would sometimes still move items if connected to Underground Belts or Loaders. (https://forums.factorio.com/35839)
    - Fixed that the multiplayer game could crash randomly due to the packet fragmentation implementation issue.
  Scripting:
    - Added LuaPlayer::disable_recipe_subgroups.
---------------------------------------------------------------------------------------------------
Version: 0.14.19
Date: 04. 11. 2016
  Bugfixes:
    - Fixed that the demo version couldn't start.
    - Fixed that items (like grenades) could be used also in the map view. (https://forums.factorio.com/35565)
    - Fixed tutorial level 2 scripting.
    - Fixed weird recipe sorting in the demo.
    - Fixed that the game crashed when it was closed while sound settings were opened.
---------------------------------------------------------------------------------------------------
Version: 0.14.18
Date: 02. 11. 2016
  Bugfixes:
    - Fixed Supply scenario Gui when playing in multiplayer.
    - Fixed unresearchable gates technology in New hope campaign level 1. (https://forums.factorio.com/35224)
    - Fixed LAN games getting deselected. (https://forums.factorio.com/35486)
    - Fixed crash when clearing items robots held through script. (https://forums.factorio.com/35585)
    - Possibly fixed xrandr related crash on Linux. (https://forums.factorio.com/35430)
  Minor features:
    - Added minimum_latency_in_ticks to server settings (for headless server) and into config (for starting game from gui).
    - Enemy players are shown on the map when the chunk is being charted (a radar or player is exposing it).
---------------------------------------------------------------------------------------------------
Version: 0.14.17
Date: 27. 10. 2016
  Bugfixes:
    - Fixed quickbar clearing wouldn't work when taking part of the stack. (https://forums.factorio.com/35211)
    - Fixed that mod browser sorting column would be ignored after using the search. (https://forums.factorio.com/35233)
    - Actually fixed Has Mods filter in browse games GUI (https://forums.factorio.com/35222).
    - Fixed that the "can't open enemy structures" error would show when it was never possible to open any GUI for the entity. (https://forums.factorio.com/35220)
    - It's now not possible to switch to the install-mods screen while there are mods pending deletion. (https://forums.factorio.com/35260)
    - Fix constant combinator ignoring item_slot_count prototype change after creation (https://forums.factorio.com/35201).
  Changes:
    - Tweaked map transfer algorithm.
---------------------------------------------------------------------------------------------------
Version: 0.14.16
Date: 25. 10. 2016
  Bugfixes:
    - Fixed crash related to technology migration. (https://forums.factorio.com/35161)
    - Fixed item sub groups wouldn't be used in the recipe GUI when item groups were disabled. (https://forums.factorio.com/35140)
    - Public and LAN game visibility are now separate settings (in GUI and in server-settings.json),
      and they can be toggled runtime with the /config command.
      See data/server-settings.example.json for a way to specify the visibility, as the old one will not work anymore.
    - Fixed has password/mods filters not working.
    - Fixed problems with determining external IP address for MP games. (https://forums.factorio.com/35192)
---------------------------------------------------------------------------------------------------
Version: 0.14.15
Date: 24. 10. 2016
  Changes:
    - Improved the browse-mods GUI by downloading mods while browsing the mods list.

items.lua
data:extend(
{
  {
    type = "item",
    name = "electric-furnace-2",
    icon = "__base__/graphics/icons/electric-furnace.png",
    flags = {"goes-to-quickbar"},
    subgroup = "smelting-machine",
    order = "c[electric-furnace-2]",
    place_result = "electric-furnace-2",
    stack_size = 50
  },
  {
    type = "ammo",
    name = "test-fast-cannon-shell",
    icon = "__base__/graphics/icons/cannon-shell.png",
    flags = {"goes-to-main-inventory"},
    ammo_type =
    {
      category = "cannon-shell",
      target_type = "direction",
      action =
      {
        type = "direct",
        action_delivery =
        {
          type = "projectile",
          projectile = "cannon-projectile",
          starting_speed = 10,
          source_effects =
          {
            type = "create-explosion",
            entity_name = "explosion-gunshot"
          },
        }
      },
    },
    subgroup = "ammo",
    order = "d[cannon-shell]-a[test-fast]",
    stack_size = 100
  },
  {
    type = "ammo",
    name = "test-decelerating-cannon-shell",
    icon = "__base__/graphics/icons/cannon-shell.png",
    flags = {"goes-to-main-inventory"},
    ammo_type =
    {
      category = "cannon-shell",
      target_type = "direction",
      action =
      {
        type = "direct",
        action_delivery =
        {
          type = "projectile",
          projectile = "test-decelerating-cannon-projectile",
          starting_speed = 1,
          source_effects =
          {
            type = "create-explosion",
            entity_name = "explosion-gunshot"
          },
        }
      },
    },
    subgroup = "ammo",
    order = "d[cannon-shell]-a[test-decelerating]",
    stack_size = 100
  },
  {
    type = "item",
    name = "roboport-2",
    icon = "__base__/graphics/icons/roboport.png",
    flags = {"goes-to-quickbar"},
    subgroup = "smelting-machine",
    order = "c[electric-furnace-2]",
    place_result = "electric-furnace-2",
    stack_size = 50
  },
  {
    type = "item-with-inventory",
    name = "portable-steel-chest",
    icon = "__base__/graphics/icons/steel-chest.png",
    flags = {"goes-to-quickbar"},
    subgroup = "storage",
    order = "a[items]-c[steel-chest]-d[portable]",
    stack_size = 1,
    inventory_size = 20,
    item_filters = {"wooden-chest"},
    item_subgroup_filters = {"belt"}
  },
  {
    type = "item-with-inventory",
    name = "portable-iron-chest",
    icon = "__base__/graphics/icons/steel-chest.png",
    flags = {"goes-to-quickbar"},
    subgroup = "storage",
    order = "a[items]-c[iron-chest]-d[portable]",
    stack_size = 1,
    inventory_size = 20,
    item_filters = {"steel-chest"},
    item_subgroup_filters = {"belt"},
    filter_mode = "blacklist"
  },
  {
    type = "item-with-inventory",
    name = "container-item",
    icon = "__base__/graphics/icons/steel-chest.png",
    flags = {"goes-to-quickbar"},
    subgroup = "storage",
    order = "a[items]-c[container]-p[portable]",
    stack_size = 1,
    inventory_size = 20
  },
})

local roboport2 = util.table.deepcopy(data.raw.item.roboport)
roboport2.name = "roboport2"
roboport2.place_result = "roboport2"
data.raw.item.roboport2 = roboport2

local labfast = util.table.deepcopy(data.raw.item.lab)
labfast.name = "labfast"
labfast.place_result = "labfast"
data.raw.item.labfast = labfast

local beacon2 = util.table.deepcopy(data.raw.item["beacon"])
beacon2.name = "beacon2"
beacon2.place_result = "beacon2"
data.raw.item.beacon2 = beacon2

local repairPack2 = util.table.deepcopy(data.raw["repair-tool"]["repair-pack"])
repairPack2.name = "repair-pack-2"
data.raw["repair-tool"]["repair-pack-2"] = repairPack2

local speedModule440 = util.table.deepcopy(data.raw["module"]["speed-module-3"])
speedModule440.name = "speed-module-440"
speedModule440.effect = { speed = {bonus = 4.4} }
data.raw["module"]["speed-module-440"] = speedModule440

local cargoWagon2 = util.table.deepcopy(data.raw["item-with-entity-data"]["cargo-wagon"])
cargoWagon2.name = "cargo-wagon-with-grid"
cargoWagon2.place_result = "cargo-wagon-with-grid"
data.raw["item-with-entity-data"][cargoWagon2.name] = cargoWagon2

local car2 = util.table.deepcopy(data.raw["item-with-entity-data"]["car"])
car2.name = "car-with-grid"
car2.place_result = "car-with-grid"
data.raw["item-with-entity-data"][car2.name] = car2

local batteryEquipment2Item = util.table.deepcopy(data.raw["item"]["battery-equipment"])
batteryEquipment2Item.name = "battery-equipment-fish"
batteryEquipment2Item.placed_as_equipment_result = "battery-equipment-fish"
data.raw["item"][batteryEquipment2Item.name] = batteryEquipment2Item

local batteryEquipment2 = util.table.deepcopy(data.raw["battery-equipment"]["battery-equipment"])
batteryEquipment2.name = "battery-equipment-fish"
batteryEquipment2.categories = {"fish"}
data.raw["battery-equipment"][batteryEquipment2.name] = batteryEquipment2

local fishEquipmentCategory = util.table.deepcopy(data.raw["equipment-category"]["armor"])
fishEquipmentCategory.name = "fish"
data.raw["equipment-category"][fishEquipmentCategory.name] = fishEquipmentCategory


recipe.lua
data:extend(
{
  {
    type = "recipe-category",
    name = "fluid-furnace"
  },
  {
    type = "recipe",
    name = "light-oil-to-petroleum-gas",
    category = "fluid-furnace",
    enabled = "true",
    energy_required = 2,
    ingredients =
    {
      {type="fluid", name="light-oil", amount=1}
    },
    results=
    {
      {type="fluid", name="petroleum-gas", amount=4}
    },
    subgroup = "fluid",
    order = "a[oil-processing]-a[advance-light-oil-processing]"
  },
  {
    type = "recipe",
    name = "heated-light-oil-to-heated-petroleum-gas",
    category = "crafting-with-fluid",
    enabled = true,
    energy_required = 1,
    ingredients = 
    {
      {type="fluid", name="light-oil", amount=1, minimum_temperature = 75, maximum_temperature = 90}
    },
    results = 
    {
      {type="fluid", name="petroleum-gas", amount=1, temperature=80}
    },
    subgroup = "fluid",
    order = "a[oil-processing]-a[heated-light-oil-processing]"
  }
})


entities.lua
data:extend(
{
  {
    type = "furnace",
    name = "electric-furnace-2",
    icon = "__base__/graphics/icons/electric-furnace.png",
    flags = {"placeable-neutral", "placeable-player", "player-creation"},
    minable = {mining_time = 1, result = "electric-furnace-2"},
    max_health = 150,
    corpse = "big-remnants",
    dying_explosion = "medium-explosion",
    light = {intensity = 1, size = 10},
    resistances =
    {
      {
        type = "fire",
        percent = 80
      }
    },
    collision_box = {{-1.2, -1.2}, {1.2, 1.2}},
    selection_box = {{-1.5, -1.5}, {1.5, 1.5}},
    module_slots = 2,
    crafting_categories = {"smelting", "fluid-furnace"},
    result_inventory_size = 0,
    crafting_speed = 2,
    energy_usage = "180kW",
    source_inventory_size = 0,
    energy_source =
    {
      type = "electric",
      usage_priority = "secondary-input",
      emissions = 0.005
    },
    working_sound =
    {
      sound =
      {
        filename = "__base__/sound/electric-furnace.ogg",
        volume = 0.7
      },
      apparent_volume = 1.5
    },
    animation =
    {
      filename = "__base__/graphics/entity/electric-furnace/electric-furnace-base.png",
      priority = "high",
      width = 129,
      height = 100,
      frame_count = 1,
      animation_speed = 0.5,
      shift = {0.421875, 0}
    },
    fluid_boxes =
    {
      {
        production_type = "input",
        pipe_picture = assembler2pipepictures(),
        pipe_covers = pipecoverspictures(),
        base_area = 10,
        base_level = -1,
        pipe_connections = {{ type="input", position = {0, -2} }}
      },
      {
        production_type = "output",
        pipe_picture = assembler2pipepictures(),
        pipe_covers = pipecoverspictures(),
        base_area = 10,
        base_level = 1,
        pipe_connections = {{ type="output", position = {0, 2} }}
      },
      off_when_no_fluid_recipe = true
    },
    fast_replaceable_group = "furnace"
  },
  {
    type = "locomotive",
    name = "turbo-locomotive",
    order = "xxx",
    icon = "__base__/graphics/icons/diesel-locomotive.png",
    flags = {"placeable-neutral", "player-creation", "placeable-off-grid", "not-on-map"},
    minable = {mining_time = 1, result = "diesel-locomotive"},
    max_health = 1000,
    corpse = "medium-remnants",
    dying_explosion = "medium-explosion",
    collision_box = {{-0.6, -2.6}, {0.6, 2.6}},
    selection_box = {{-0.85, -2.6}, {0.9, 2.5}},
    drawing_box = {{-1, -4}, {1, 3}},
    weight = 2000,
    max_speed = 12,
    max_power = "50000kW",
    reversing_power_modifier = 0.6,
    braking_force = 10000,
    friction_force = 0.0015,
    vertical_selection_shift = -0.5,
    -- this is a percentage of current speed that will be subtracted
    air_resistance = 0.002,
    connection_distance = 3.3,
    joint_distance = 4.6,
    energy_per_hit_point = 5,
    resistances =
    {
      {
        type = "fire",
        decrease = 15,
        percent = 50
      },
      {
        type = "physical",
        decrease = 15,
        percent = 30
      },
      {
        type = "impact",
        decrease = 50,
        percent = 60
      },
      {
        type = "explosion",
        decrease = 15,
        percent = 30
      },
      {
        type = "acid",
        decrease = 10,
        percent = 20
      }
    },
    energy_source =
    {
      type = "burner",
      effectivity = 1000,
      fuel_inventory_size = 3,
      smoke =
      {
        {
          name = "smoke",
          deviation = {0.1, 0.1},
          frequency = 210,
          position = {0, 0},
          slow_down_factor = 3,
          starting_frame = 1,
          starting_frame_deviation = 5,
          starting_frame_speed = 0,
          starting_frame_speed_deviation = 5,
          height = 2,
          height_deviation = 0.2,
          starting_vertical_speed = 0.2,
          starting_vertical_speed_deviation = 0.06,
        }
      }
    },
    front_light =
    {
      {
        type = "oriented",
        minimum_darkness = 0.3,
        picture =
        {
          filename = "__core__/graphics/light-cone.png",
          priority = "medium",
          scale = 2,
          width = 200,
          height = 200
        },
        shift = {-0.6, -16},
        size = 2,
        intensity = 0.6
      },
      {
        type = "oriented",
        minimum_darkness = 0.3,
        picture =
        {
          filename = "__core__/graphics/light-cone.png",
          priority = "medium",
          scale = 2,
          width = 200,
          height = 200
        },
        shift = {0.6, -16},
        size = 2,
        intensity = 0.6
      }
    },
    back_light = rolling_stock_back_light(),
    stand_by_light = rolling_stock_stand_by_light(),
    pictures =
    {
      priority = "very-low",
      width = 238,
      height = 234,
      direction_count = 256,
      filenames =
      {
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-01.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-02.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-03.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-04.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-05.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-06.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-07.png",
        "__base__/graphics/entity/diesel-locomotive/diesel-locomotive-08.png"
      },
      line_length = 4,
      lines_per_file = 8,
      shift = {0.0, -0.5}
    },

    wheels =
    {
      priority = "very-low",
      width = 115,
      height = 115,
      direction_count = 256,
      filenames =
      {
        "__base__/graphics/entity/diesel-locomotive/train-wheels-01.png",
        "__base__/graphics/entity/diesel-locomotive/train-wheels-02.png"
      },
      line_length = 8,
      lines_per_file = 16
    },

    rail_category = "regular",

    stop_trigger =
    {
      -- left side
      {
        type = "create-smoke",
        repeat_count = 125,
        entity_name = "smoke-train-stop",
        initial_height = 0,
        -- smoke goes to the left
        speed = {-0.03, 0},
        speed_multiplier = 0.75,
        speed_multiplier_deviation = 1.1,
        offset_deviation = {{-0.75, -2.7}, {-0.3, 2.7}}
      },
      -- right side
      {
        type = "create-smoke",
        repeat_count = 125,
        entity_name = "smoke-train-stop",
        initial_height = 0,
        -- smoke goes to the right
        speed = {0.03, 0},
        speed_multiplier = 0.75,
        speed_multiplier_deviation = 1.1,
        offset_deviation = {{0.3, -2.7}, {0.75, 2.7}}
      },
      {
        type = "play-sound",
        sound =
        {
          {
            filename = "__base__/sound/train-breaks.ogg",
            volume = 0.6
          },
        }
      },
    },
    drive_over_tie_trigger = drive_over_tie(),
    tie_distance = 50,
    crash_trigger = crash_trigger(),
    working_sound =
    {
      sound =
      {
        filename = "__base__/sound/train-engine.ogg",
        volume = 0.4
      },
      match_speed_to_activity = true,
    },
    open_sound = { filename = "__base__/sound/car-door-open.ogg", volume=0.7 },
    close_sound = { filename = "__base__/sound/car-door-close.ogg", volume = 0.7 },
    sound_minimum_speed = 0.5;
  },
  {
    type = "corpse",
    name = "small-biter-fast-corpse",
    icon = "__base__/graphics/icons/small-biter-corpse.png",
    selection_box = {{-0.8, -0.8}, {0.8, 0.8}},
    selectable_in_game = false,
    subgroup="corpses",
    order = "c[corpse]-a[biter]-a[small]",
    flags = {"placeable-neutral", "placeable-off-grid", "building-direction-8-way", "not-repairable", "not-on-map"},
    dying_speed = 0.04,
    time_before_removed = 60,
    final_render_layer = "corpse",
    animation = biterdieanimation(smallbiterscale, small_biter_tint1, small_biter_tint2)
  },
  {
    type = "decorative",
    name = "decorative-with-object-layer-collisions",
    flags = {"placeable-neutral", "not-on-map"},
    collision_mask = { "item-layer", "object-layer", "player-layer", "water-tile"},
    icon = "__base__/graphics/icons/small-stone-rock.png",
    selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
    collision_box = {{-0.4, -0.4}, {0.4, 0.4}},
    render_layer = "decorative",
    order = "z",
    pictures =
    {
      {
        filename = "__base__/graphics/entity/decorative/small-stone-rock/small-stone-rock-01.png",
        width = 47,
        height = 37,
        shift = {0.21, -0.18},
        tint = {r=1, g=0, b=0, a=1}
      }
    },
    vehicle_impact_sound =  { filename = "__base__/sound/car-stone-impact.ogg", volume = 1.0 }
  },
  {
    type = "electric-energy-interface",
    name = "electric-energy-interface-test",
    icon = "__base__/graphics/icons/small-biter-corpse.png",
    flags = {"placeable-neutral", "player-creation"},
    max_health = 500,
    corpse = "small-remnants",
    collision_box = {{-0.35, -0.35}, {0.35, 0.35}},
    selection_box = {{-0.5, -0.5}, {0.5, 0.5}},
    energy_source =
    {
      buffer_capacity = "50MJ",
      usage_priority = "primary-input",
      input_flow_limit = "1500kW",
      output_flow_limit = "0kW"
    },
    order="b[bomb]",
    subgroup = "equipment"
  }
})

local roboport2 = util.table.deepcopy(data.raw.roboport.roboport)
roboport2.name = "roboport2"
roboport2.logistics_radius = 50
roboport2.minable.result = "roboport2"
roboport2.fast_replaceable_group = "roboport"
data.raw.roboport.roboport2 = roboport2

data.raw.roboport.roboport.fast_replaceable_group = "roboport"

local labfast = util.table.deepcopy(data.raw.lab.lab)
labfast.name = "labfast"
labfast.researching_speed = 2
data.raw.lab.labfast = labfast

local beacon2 = util.table.deepcopy(data.raw.beacon["beacon"])
beacon2.name = "beacon2"
beacon2.place_result = "beacon2"
beacon2.minable.result = "beacon2"
beacon2.fast_replaceable_group = "beacon"
data.raw.beacon.beacon2 = beacon2

local cannon_projectile2 = util.table.deepcopy(data.raw.projectile["cannon-projectile"])
cannon_projectile2.name = "test-decelerating-cannon-projectile"
cannon_projectile2.acceleration = -0.1
data.raw.projectile["test-decelerating-cannon-projectile"] = cannon_projectile2


data.raw.beacon["beacon"].fast_replaceable_group = "beacon"

local fastCorpse = util.table.deepcopy(data.raw.corpse["big-biter-corpse"])
fastCorpse.name = "fast-corpse"
fastCorpse.time_before_removed = 10
fastCorpse.dying_speed = 1
data.raw.corpse["fast-corpse"] = fastCorpse

data.raw["boiler"]["boiler"].additional_pastable_entities = {"pipe"}


local cargoWagon2 = util.table.deepcopy(data.raw["cargo-wagon"]["cargo-wagon"])
cargoWagon2.name = "cargo-wagon-with-grid"
cargoWagon2.equipment_grid = "large-equipment-grid"
cargoWagon2.minable = {mining_time = 1, result = cargoWagon2.name}
data.raw["cargo-wagon"][cargoWagon2.name] = cargoWagon2

local car2 = util.table.deepcopy(data.raw.car["car"])
car2.name = "car-with-grid"
car2.equipment_grid = "large-equipment-grid"
car2.minable = {mining_time = 1, result = car2.name}
data.raw.car[car2.name] = car2
